homeassistant/components/minecraft_server/__init__.py

```diff
@@ -28,9 +32,19 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+def load_dnspython_rdata_classes() -> None:
+    """Load dnspython rdata classes used by mcstatus."""
+    for rdtype in dns.rdatatype.RdataType:
+        if not dns.rdatatype.is_metatype(rdtype) or rdtype == dns.rdatatype.OPT:
+            dns.rdata.get_rdata_class(dns.rdataclass.IN, rdtype)  # type: ignore[no-untyped-call]
+
```

Comment:
Shouldn't this be handled in the client library?

Reply:
This was also my wish, but the dev won't fix it 😢 See the disussion in my issue report here: [https://github.com/rthalley/dnspython/issues/1083](https://github.com/rthalley/dnspython/issues/1083)

BTW: bdraco was also involved in the discussions and was for the workaround within my integration.

Reply:
Sure, but that is the dnspython library. This integration uses a library in between which could do this.

Reply:
Ah sorry, I misunderstood you. Yes theoretically it could be handled within mcstatus. But as far as I know, they don't use the executor. So not sure how they could do it.

Reply:
I just talked to one of the devs of mcstatus and they asked us to handle it on our side as it is an "edge case" from their point of view.

But I also got to know that the workaround for dnspython I added in this PR will be part of the next release of dnspython ([see this PR](https://github.com/rthalley/dnspython/pull/1095)). Which means that I can remove the workaround later and call it from dnspython directly.

So if you don't mind I would like to continue this PR as it is and replace it with the "official" API from dnspython later once version 2.7.0 is released (which could take some time looking at the release frequency in the past).